lib/browser/mixins/browser_wait.ts

```diff
@@ -46,7 +49,10 @@ export default abstract class BrowserWait extends BrowserNavigation {
                 return !WendigoUtils.isVisible(element);
             }, timeout, selector);
         } catch (err) {
-            throw new TimeoutError("waitUntilNotVisible", `Waiting for element "${selector}" to not be visible`, timeout);
+            if (err instanceof TimeoutError) {
+                throw new TimeoutError("waitUntilNotVisible", `Waiting for element "${selector}" to not be visible`, timeout);
```

Comment:
Consider using @overrideError decorator (I'm not sure why I didn't added it here, maybe there was some reason I don't see now, if the decorator doesn't work, this approach seems good enough